gameDriver.py

```diff
@@ -0,0 +1,187 @@
+import scrabbler as sc
+import random
+import time
+
+LETTERS = """\
+AAAAAAAAAB\
+BCCDDDDEEE\
+EEEEEEEEEF\
+FGGGHHIIII\
+IIIIIJKLLL\
+LMMNNNNNNO\
+OOOOOOOPPQ\
+RRRRRRSSSS\
+TTTTTTUUUU\
+VVWWXYYZ??\
+"""
+
+ALREADY_ENDGAME = False
+
+def get_full_bag():
+    """Returns a list of letters in the whole bag."""
+    return list(LETTERS)
+
+
+def generate_rack_and_bag(rack, bag):
+    """Randomly chooses tiles from bag and places in rack"""
+    for i in range(7 - len(rack)):
+        # If bag has ended then end game begins (as of right now this 
+        # doesn't formally mean anything, just print statement)
+        if not bag:
+            global ALREADY_ENDGAME
+            if not ALREADY_ENDGAME:
+                print('|||||||||||||||||||| END GAME STARTS NOW ||||||||||||||||||||')
+                ALREADY_ENDGAME = True
+            return rack, bag
+        new_tile = random.choice(bag)
+        rack.append(new_tile)
+        bag.remove(new_tile)
+
+    return rack, bag
+
+
+def main():
+    # List of letters we can still pick from.
+    bag = get_full_bag()
+
+    # Rack starts out empty.
+    player1Turn = True
+    player1Rack = []
+    player2Rack = []
+    player1Score = 0
+    player2Score = 0
+
+    # Initialize game and board
+    game = sc.Game()
+    board = game.get_board()
+
+    times = []
+
+    # Keep playing until we're out of tiles or solutions.
+    while True:
+        if player1Turn:
```

Comment:
Can you make all of the code below that executes during a player's turn into a helper function that takes in the current player as a parameter? All of this code is basically duplicated for each player.

Reply:
I'll do this in a follow up pr.